### PR TITLE
pizza sharpener (sharpens pizzas)

### PR DESCRIPTION
**Pull request by BenLubar**
[FEATURE]

## About the PR

It's a pizza sharpener! It appears to be a normal pizza cutter, but it has a secret switch that activates sharpening mode. In sharpening mode, it... uh... sharpens pizzas.

Sharpened pizzas and sharpened pizza slices cause bleeding if you eat them or get hit by them, and can be used as throwing weapons. A whole pizza will cut someone in half if thrown enough times, and a slice will embed itself in the chest of your victim, transferring whatever crap was in it directly into their bloodstream.

Cutting a sharpened pizza creates sharpened pizza slices.

You can also stuff it into a pizza vending machine, which will make the pizza vending machine sharpen and throw its pizzas, and also significantly reduce the cooking time.

There's no way to tell whether pizza has been sharpened, except for the fact that the chef is throwing pizzas at people and they're bleeding all over the floor.

## Why's this needed?

The words "pizza sharpener" came to me in a dream and now everyone must suffer.

## Changelog

```changelog
(u)BenLubar:
(*)Added a new Chef traitor item: Pizza Sharpener.
```

### CODE REVIEW

**Comment by BatElite** [comment]
Now that's cutting edge traitoring!

NB you misspelled pizza in the buylist description.

**Comment by BenLubar** [comment]
> NB you misspelled pizza in the buylist description.

wait what where

File: code/datums/syndicate_buylist.dm [hunk 1]
```diff
@@ -707,6 +707,13 @@ This is basically useless for anyone but miners.
 	job = list("Medical Director")
 	blockedmode = list(/datum/game_mode/spy, /datum/game_mode/revolution)
 
+/datum/syndicate_buylist/traitor/pizza_sharpener
+	name = "Pizza Sharpener"
+	item = /obj/item/kitchen/utensil/knife/pizza_cutter/traitor
+	cost = 5
+	desc = "Have you ever been making a piza and thought \"this pizza would be better if I could fatally injure someone by throwing it at them\"? Well think no longer! Because you're sharpening pizzas now. You weirdo."
+	job = list("Chef")
+
```
Comment (flappybatpal) [inline code/datums/syndicate_buylist.dm:716]: not_in_crates = 1
blockedmode = list(/datum/game_mode/revolution)

**Review by flappybatpal — CHANGES_REQUESTED** [review]
1. This is a specialist 5 cost item that would use up a chunk of crate TC.
2. It would currently show up in revolutionary buy lists.

File: code/modules/food_and_drink/snacks.dm [hunk 2]
```diff
@@ -89,7 +90,13 @@
 				src.visible_message("<b>[src]</b> <i>says, \"I'm pizza.\"</i>")
 
 	attackby(obj/item/W as obj, mob/user as mob)
-		if (istool(W, TOOL_CUTTING | TOOL_SAWING))
+		if (istype(W, /obj/item/kitchen/utensil/knife/pizza_cutter/traitor) && W:sharpener_mode)
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:93]: `:` bad

File: code/modules/food_and_drink/snacks.dm [hunk 3]
```diff
@@ -68,6 +68,7 @@
 	amount = 6
 	heal_amt = 3
 	var/topping_color = null
+	var/sharpened = 0
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:71]: boolean - change to FALSE or TRUE across file

File: code/modules/food_and_drink/snacks.dm [hunk 4]
```diff
@@ -128,8 +139,50 @@
 				user.visible_message("<span class='alert'><b>[user]</b> futilely attempts to shove [src] into [M]'s mouth!</span>")
 				return
 		else
+			if (sharpened)
+				boutput(M, "<span class='alert'>The pizza was too pointy!</span>")
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:143]: both this and the above pointy hurt can both occur, why??

File: code/modules/food_and_drink/snacks.dm [hunk 5]
```diff
@@ -128,8 +139,50 @@
...
+				take_bleeding_damage(M, user, 50, DAMAGE_CUT)
+			..()
+
+	attack_self(var/mob/user as mob)
+		if (sharpened && prob(15))
+			boutput(user, "<span class='alert'>The pizza was too pointy!</span>")
+			take_bleeding_damage(user, null, 15, DAMAGE_CUT)
+		if (!src.sliced)
+			boutput(user, "<span class='alert'>You can't just cram that in your mouth, you greedy beast!</span>")
+			user.visible_message("<b>[user]</b> stares at [src] in a confused manner.")
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:153]: why is this here...

File: code/modules/food_and_drink/snacks.dm [hunk 6]
```diff
@@ -128,8 +139,50 @@
...
+			return
+		else
+			if (sharpened)
+				boutput(user, "<span class='alert'>The pizza was too pointy!</span>")
+				take_bleeding_damage(user, user, 50, DAMAGE_CUT)
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:158]: ditto double occur

File: code/modules/food_and_drink/snacks.dm [hunk 7]
```diff
@@ -128,8 +139,50 @@
...
 			..()
 
+	throw_impact(M)
+		..()
+		if (!sharpened || M == null)
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:163]: isnull

File: code/modules/food_and_drink/snacks.dm [hunk 8]
```diff
@@ -128,8 +139,50 @@
...
+				src.set_loc(M)
+				src.transfer_all_reagents(M)
+			random_brute_damage(M, 11)
+			take_bleeding_damage(M, null, 25, DAMAGE_STAB)
+		else
+			playsound(src.loc, "sound/weapons/slashcut.ogg", 100, 1)
+			if (ishuman(M))
+				var/mob/living/carbon/human/H = M
+				H.changeStatus("weakened", 5 SECONDS)
+				H.TakeDamage("chest", prob(5) ? 50 : 15, 0, 0, DAMAGE_CUT)
```
Comment (ZeWaka) [inline code/modules/food_and_drink/snacks.dm:181]: high variability in damage - not that fun imo

File: code/modules/vending/vending.dm [hunk 9]
```diff
@@ -1581,11 +1581,22 @@
 	slogan_list = list("A revolution in the pizza industry!",
 	"Prepared in moments!",
 	"I'm a chef who works 24 hours a day!")
+	var/sharpen = 0
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:1584]: TRUE FALSE ditto

File: code/modules/vending/vending.dm [hunk 10]
```diff
@@ -1622,13 +1633,17 @@
 					icon_state = "pizza-vend"
 					src.generate_HTML(1)
 					updateUsrDialog()
-					sleep(20 SECONDS)
+					if (!src.sharpen)
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:1636]: why

File: code/modules/vending/vending.dm [hunk 11]
```diff
@@ -1581,11 +1581,22 @@
...
 
 	lr = 1
 	lg = 0.6
 	lb = 0.2
 
+	attackby(obj/item/W, mob/user)
+		if (!sharpen && istype(W, /obj/item/kitchen/utensil/knife/pizza_cutter/traitor))
+			sharpen = 1
+			add_fingerprint(user)
+			boutput(user, "You jam the pizza sharpener inside the vending machine.")
```
Comment (ZeWaka) [inline code/modules/vending/vending.dm:1594]: this whole mechanic is pretty strong, easily farmable pre-sharpened pizzas

File: code/obj/item/kitchen.dm [hunk 12]
```diff
@@ -424,6 +424,13 @@ TRAYS
 		user.TakeDamage("head", 150, 0)
 		return 1
 
+/obj/item/kitchen/utensil/knife/pizza_cutter/traitor
+	var/sharpener_mode = 0
```
Comment (ZeWaka) [inline code/obj/item/kitchen.dm:428]: TRUE FALSE

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drink/snacks.dm (hunk 5 above)
Comment (BenLubar) [reply code/modules/food_and_drink/snacks.dm:153]: Pizzas have this code for when you're trying to eat by clicking yourself but not for when you click the item in your hand. I added it for consistency.

File: code/modules/food_and_drink/snacks.dm (hunk 6 above)
Comment (BenLubar) [reply code/modules/food_and_drink/snacks.dm:158]: Eating the pizza hurts a lot, but trying to feed it to someone also has a chance of causing bleeding.

File: code/modules/food_and_drink/snacks.dm (hunk 4 above)
Comment (BenLubar) [reply code/modules/food_and_drink/snacks.dm:143]: Eating the pizza hurts a lot, but trying to feed it to someone also has a chance of causing bleeding.

File: code/modules/vending/vending.dm (hunk 10 above)
Comment (BenLubar) [reply code/modules/vending/vending.dm:1636]: If it's not sped up, the pizza vending machine would spit out a pizza at someone's position 20 seconds later.

File: code/modules/vending/vending.dm (hunk 11 above)
Comment (BenLubar) [reply code/modules/vending/vending.dm:1594]: Sharpened pizzas are probably strongest with fun reagents in them, and the pizzas from the vending machine can't do that.

File: code/modules/vending/vending.dm (hunk 10 above)
Comment (ZeWaka) [reply code/modules/vending/vending.dm:1636]: but why only for the not sharpened case?

File: code/modules/vending/vending.dm (hunk 11 above)
Comment (ZeWaka) [reply code/modules/vending/vending.dm:1594]: can you not put reagents on the pizza afterwards, like all food items?

Comment (BenLubar) [reply code/modules/vending/vending.dm:1594]: I'm pretty sure syringes only allow you to put reagents into food items that don't have their own method of adding reagents.

File: code/modules/vending/vending.dm (hunk 10 above)
Comment (BenLubar) [reply code/modules/vending/vending.dm:1636]: The 20 second delay is for the cooking for regular pizza vending machines, and [as of the most recent commit] after the vending for sharpened pizza vending machines.

File: code/modules/food_and_drink/snacks.dm [hunk 13]
```diff
@@ -128,15 +142,57 @@
...
 			..()
 
+	throw_impact(M)
+		..()
+		if (!sharpened || isnull(M))
+			return
+		if (sliced)
+			if (ishuman(M))
+				var/mob/living/carbon/human/H = M
+				H.implant.Add(src)
```
Comment (pali6) [inline code/modules/food_and_drink/snacks.dm:171]: I'm not sure whether `implant` doesn't expect only `/obj/item/implant` to be in it. Probably not but worth double checking maybe?

File: code/modules/vending/vending.dm (hunk 11 above)
Comment (pali6) [reply code/modules/vending/vending.dm:1594]: Syringes allow you to inject pizzas just fine. If you want to prevent that you'll need to override `is_open_container` to return `0` probably (though `is_open_container` is awful so even that might not work).

File: code/modules/food_and_drink/snacks.dm (hunk 13 above)
Comment (BenLubar) [reply code/modules/food_and_drink/snacks.dm:171]: I used the code for throwing knives as a reference.

File: code/modules/vending/vending.dm (hunk 11 above)
Comment (BenLubar) [reply code/modules/vending/vending.dm:1594]: In any case, if you try to get the pizzas from the sabotaged vending machine, they're getting launched at you

**Comment by BenLubar** [comment]
reopened by popular demand (two people asked about it)

**Review by pali6 — APPROVED** [review]
Eh if the syringe thing proves too abusable we'll jsut make pizza non-injectable or something.
